test/lib/time.rb

```diff
@@ -71,6 +85,21 @@ def minutes_conversion(minutes)
     "#{hours}:#{rest}"
   end
 
+  def inner_sections?(minutes_arr, arr, time)
+    minutes_arr[arr][time + 1] >= minutes_arr[arr + 1][time + 1] && minutes_arr[arr][time] <= minutes_arr[arr + 1][time]
+  end
+
+  def delete_inner_sector(minutes_arr)
+    (0..minutes_arr.length).each do |arr|
+      (0..1).each do |time|
+        return minutes_arr unless inner_sections?(minutes_arr, arr, time)
+
+        minutes_arr.delete_at(1)
```

Comment:
Why delete at 1?

Reply:
because the second section time is in the first section time. And first section time is a second array.